### ISSUE TITLE
[BUG] User not added to group

### ISSUE THREAD

**Issue by mraufer** [code-ready/crc#2621]
### General information

  * OS: Windows
  * Hypervisor: Hyper-V
  * Did you run `crc setup` before starting it (Yes/No)? no
  * Running CRC on: Laptop

## CRC version
```bash
# Put `crc version` output here
``` 
CodeReady Containers version: 1.30.1+376cc3c1
OpenShift version: 4.8.2 (not embedded in executable)
  
## CRC status
```bash
# Put `crc status --log-level debug` output here
```
DEBU CodeReady Containers version: 1.30.1+376cc3c1
DEBU OpenShift version: 4.8.2 (not embedded in executable)
DEBU Running 'crc setup'
INFO Checking if crc-admin-helper executable is cached
INFO Checking for obsolete admin-helper executable
DEBU Checking if an older admin-helper executable is installed
DEBU No older admin-helper executable found
INFO Checking if running on a supported CPU architecture
INFO Checking minimum RAM requirements
DEBU Total memory of system is 34081480704 bytes
INFO Checking if running in a shell with administrator rights
DEBU Running '$currentPrincipal = New-Object Security.Principal.WindowsPrincipal([Security.Principal.WindowsIdentity]::GetCurrent());$currentPrincipal.IsInRole([Security.Principal.WindowsBuiltInRole]::Administrator)'
INFO Checking Windows 10 release
DEBU Running '(Get-ItemProperty -Path "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion" -Name ReleaseId).ReleaseId'
INFO Checking Windows edition
DEBU Running '(Get-ItemProperty -Path "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion").EditionID'
DEBU Running on Windows Enterprise edition
INFO Checking if Hyper-V is installed and operational
DEBU Running '@(Get-Wmiobject Win32_ComputerSystem).HypervisorPresent'
DEBU Running '@(Get-Service vmms).Status'
INFO Checking if crc-users group exists
DEBU Running 'Get-LocalGroup -Name crc-users'
INFO Checking if current user is in Hyper-V group and crc-users group
DEBU Running 'Get-LocalGroupMember -Name crc-users -Member '0123456789''
DEBU Command failed: exit status 1
DEBU stdout:
DEBU stderr: Get-LocalGroupMember : Principal 0123456789 was not found.
At line:1 char:43
+ ... lyContinue'; Get-LocalGroupMember -Name crc-users -Member '0123456789'
+                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (0123456789:String) [Get-LocalGroupMember], PrincipalNotFoundException
    + FullyQualifiedErrorId : PrincipalNotFound,Microsoft.PowerShell.Commands.GetLocalGroupMemberCommand

DEBU exit status 1
INFO Adding current user to group
DEBU Running 'Get-LocalGroupMember -Name crc-users -Member '0123456789''
DEBU Command failed: exit status 1
DEBU stdout:
DEBU stderr: Get-LocalGroupMember : Principal 0123456789 was not found.
At line:1 char:43
+ ... lyContinue'; Get-LocalGroupMember -Name crc-users -Member '0123456789'
+                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (0123456789:String) [Get-LocalGroupMember], PrincipalNotFoundException
    + FullyQualifiedErrorId : PrincipalNotFound,Microsoft.PowerShell.Commands.GetLocalGroupMemberCommand

INFO Will run as admin: adding current user to crc-users group
DEBU Running 'C:\Users\XXXXXX~1\AppData\Local\Temp\crcScripts190497199\runAsAdmin.ps1'
DEBU Running '$sid = New-Object System.Security.Principal.SecurityIdentifier("S-1-5-32-578")
        @([Security.Principal.WindowsPrincipal][Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole($sid)'
Please reboot your system and run 'crc setup' to complete the setup process

## OS version
```bash
# Put `systeminfo` output here
``` 
OS Name:                   Microsoft Windows 10 Enterprise
OS Version:                10.0.19042 N/A Build 19042

### Steps to reproduce

  1.  run Windows MSI installer, fix issue #2620 with workaround #2581 -> 2620 should NOT have been closed because the problem did not go away, the workaround may have caused this here or other problems!
  2. run crc setup
  3. reboot as instructed
  4. loop - crc setup never completes because I am not logged in as a local user (0123456789) but as a domain/cloud user (something\0123456789) - crc just does not get this, cannot find a non-existing local user, and keeps messing with local groups that it did not need before ...

### Expected
completion of crc setup without any issues

### Actual
loop, see above

### Logs
see above

**Comment by mraufer** [comment]
funny enough, crc setup found the full user (something\0123456789) the first time I ran it and added it to the crc-users local group that it had created (net localgroup crc-users lists it correctly). But crc setup cannot find that user again at any subsequent run. Is it looking for something different than before? A simple whoami would get it the right user.

**Comment by anjannath** [comment]
@mraufer What is now the output of `crc setup --check-only` ?

**Comment by mraufer** [comment]
output below as requested, plus user listing of relevant groups and local users:

crc setup --check-only
INFO Checking if crc-admin-helper executable is cached
INFO Checking for obsolete admin-helper executable
INFO Checking if running on a supported CPU architecture
INFO Checking minimum RAM requirements
INFO Checking if running in a shell with administrator rights
INFO Checking Windows 10 release
INFO Checking Windows edition
INFO Checking if Hyper-V is installed and operational
INFO Checking if crc-users group exists
INFO Checking if current user is in Hyper-V group and crc-users group
exit status 1

whoami
something\0123456789

net localgroup "hyper-v administrators"
Alias name     hyper-v administrators
Comment        <localized comment>.
Members
-------------------------------------------------------------------------------
Administrator
something\0123456789
The command completed successfully.

net localgroup crc-users
Alias name     crc-users
Comment        Group for CodeReady Containers users
Members
-------------------------------------------------------------------------------
something\0123456789
The command completed successfully.

net user
User accounts for \\HOSTNAME
-------------------------------------------------------------------------------
Administrator            DefaultAccount           defaultuser100000
Guest                     WDAGUtilityAccount
The command completed successfully.

**Comment by gbraad** [comment]
Duplicate of #2620 #2581

**Comment by jackhu008** [comment]
This is does not look like a duplicate of #2581 to me.  #2581 is a locale issue.

[#2446](https://github.com/crc-org/crc/issues/2446) provides a solution: reset your local USERNAME environment variable to include the domain prefix:

C:\...> set USERNAME=something\0123456789
C:\...> crc setup --log-level debug

Also note:  the commands need to run in a console/shell without admin privileges.  Otherwise another installation error would occur (scheduled task).

**Comment by anjannath** [comment]
@jackhu008 this should be fixed in newer CRC releases, if you still face this please open a new issue.

**Comment by mmurhamm** [comment]
@jackhu008, your suggested solution actually works, but only from command line. The graphical setup that starts after reboot does not recognize this "trick" and fails every time. If you can work without the taskbar feature, then you are good to go. However, not a nice fix, why cannot the setup tool do the same or ask for a prefix?
